pkg/frontend/querymiddleware/spin_off_subqueries_queryable.go

```diff
@@ -110,62 +110,64 @@ func (q *spinOffSubqueriesQuerier) Select(ctx context.Context, _ bool, hints *st
 		if err != nil {
 			return storage.ErrSeriesSet(errors.Wrap(err, "failed to parse subquery step"))
 		}
+		var queryOffset time.Duration
+		if offsetStr == "" {
+			queryOffset = 0
+		} else if queryOffset, err = time.ParseDuration(offsetStr); err != nil {
+			return storage.ErrSeriesSet(errors.Wrap(err, "failed to parse subquery offset"))
+		}
 
-		end := q.req.GetEnd()
-		// Align query to absolute time by querying slightly into the future.
+		end := q.req.GetEnd() - queryOffset.Milliseconds()
+		// Subqueries are aligned on absolute time, while range queries are aligned on relative time.
```

Comment:
[nit] This seems clearer to me:

```suggestion
		// Subqueries are always aligned to absolute time in PromQL, so we need to make the same adjustment here for correctness.
```